### PR TITLE
Mm/add enrollment status to site

### PR DESCRIPTION
**Pull request by Seeker1911**

### CODE REVIEW

File: rest-api/tools/import_organizations.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
 SITE_NOTES_COLUMN = 'Notes'
 # TODO: switch this back to 'Status' [DA-538]
 SITE_STATUS_COLUMN = 'PTSC Scheduling Status'
+ENROLLING_STATUS_COLUMN = 'Actively Enrolling Status'
```
Comment (danqrodney) [inline rest-api/tools/import_organizations.py]: Let's ask Ashley what she wants to call this.

File: rest-api/tools/import_organizations.py [hunk 2]
```diff
@@ -152,6 +153,7 @@ def _entity_from_row(self, row):
     except TypeError:
       logging.warn('Invalid site status %s for site %s', row[SITE_STATUS_COLUMN], google_group)
       return None
+    enrolling_status = EnrollingStatus(row[ENROLLING_STATUS_COLUMN].upper())
```
Comment (danqrodney) [inline rest-api/tools/import_organizations.py]: If you're going to require this column to be present in the data, you should add it to the array pf required fields passed into the super constructor on line 117.

However we need to make sure that field actually gets added before we can do this.

You may also want to catch TypeError here the way we did for SiteStatus (in case we get some value other than ACTIVE or INACTIVE.)

**Comment by danqrodney** [comment]
While you're messing with import_organizations.py, it would be good to make it send all awardee and organization IDs to upper case before importing, the way we currently send site IDs to lowercase. (Nit.)

File: rest-api/dao/site_dao.py [hunk 3]
```diff
@@ -19,6 +19,7 @@ class _FhirSite(FhirMixin, BackboneElement):
     FhirProperty('display_name', str, required=True),
     FhirProperty('mayolink_client_number', long),
     FhirProperty('site_status', str, required=True),
+    FhirProperty('enrolling_status', str, required=True),
```
Comment (danqrodney) [inline rest-api/dao/site_dao.py]: If you're going to make this required, we have to make sure to update the data in the database to have this column populated for all sites before we push these changes out... so like in the Alembic migration, set enrollingStatus to INACTIVE for all existing sites or something.

Alternatively, make this not required, and handle the None case below.

File: rest-api/tools/import_organizations.py (hunk 1 above)
Comment (Seeker1911) [reply rest-api/tools/import_organizations.py]: I've reached out, still waiting on response.

File: rest-api/tools/import_organizations.py (hunk 2 above)
Comment (Seeker1911) [reply rest-api/tools/import_organizations.py]: done

File: rest-api/tools/import_organizations.py (hunk 1 above)
Comment (danqrodney) [reply rest-api/tools/import_organizations.py]: OK. For now, just call this "Status" to match what's in the sheet already?

File: rest-api/dao/site_dao.py (hunk 3 above)
Comment (Seeker1911) [reply rest-api/dao/site_dao.py]: done

File: rest-api/tools/import_organizations.py (hunk 1 above)
Comment (Seeker1911) [reply rest-api/tools/import_organizations.py]: done

File: README.md [hunk 4]
```diff
@@ -390,15 +396,15 @@ Example response:
 * Authorized users can generate API keys for access.
 * Awardees are responsible for rotating keys on a three day timeframe.
     ** Permissions will be revoked after this time.
-* Awardee must specify Awardee or Organization/Site of Awardee in call to API.
+* Awardee must specify Awardee  in call to API.
```
Comment (danqrodney) [inline README.md]: I would rephrase this as:

Service accounts for specific awardees must specify the "awardee" parameter in their requests.

File: rest-api/test/unit_test/api_test/awardee_api_test.py [hunk 5]
```diff
@@ -98,13 +98,15 @@ def test_get_awardee_with_organizations_active_only(self):
...
              'displayName': 'Zebras Rock',
+             'enrollingStatus': 'INACTIVE',
              'siteStatus': 'INACTIVE',
              'address': {}
             },
              {'id': 'hpo-site-1',
               'displayName': 'Site 1',
               'mayolinkClientNumber': 123456,
               'siteStatus': 'ACTIVE',
+              'enrollingStatus': 'ACTIVE',
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/awardee_api_test.py]: You might want a test in here for when enrollingStatus isn't set.

File: rest-api/test/unit_test/api_test/awardee_api_test.py [hunk 6]
```diff
@@ -95,6 +95,12 @@ def test_get_awardee_with_organizations_active_only(self):
     # we are now filtering out 'INACTIVE' sites by default.
     self.assertEqual(self._make_expected_pitt_awardee_resource(), result)
 
+  def test_get_not_enrolling_awardees_with_organizations(self):
+    self._setup_unset_enrollment_site()
+    result = self.send_get('Awardee?_inactive=true')
+    print result
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/awardee_api_test.py]: Remove the "print" please.

File: rest-api/tools/import_organizations.py [hunk 7]
```diff
@@ -124,7 +124,8 @@ def __init__(self):
 
   def _entity_from_row(self, row):
     google_group = row[SITE_SITE_ID_COLUMN].lower()
-    organization = self.organization_dao.get_by_external_id(row[SITE_ORGANIZATION_ID_COLUMN])
+    organization = self.organization_dao.get_by_external_id(
+                                        row[SITE_ORGANIZATION_ID_COLUMN].upper())
```
Comment (danqrodney) [inline rest-api/tools/import_organizations.py]: You can't just change it in these places. You have to call upper() everywhere that awardee ID and organization ID are used.

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/tools/import_organizations.py#L80
https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/tools/import_organizations.py#L102

etc.

File: README.md [hunk 8]
```diff
@@ -396,7 +396,7 @@ Example response:
 * Authorized users can generate API keys for access.
 * Awardees are responsible for rotating keys on a three day timeframe.
     ** Permissions will be revoked after this time.
-* Awardee must specify Awardee  in call to API.
+* Service account for specific awardees  must specify the awardee parameter in requests.
```
Comment (danqrodney) [inline README.md]: "Service accounts"

File: README.md (hunk 4 above)
Comment (Seeker1911) [reply README.md]: done

File: rest-api/test/unit_test/api_test/awardee_api_test.py (hunk 5 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/awardee_api_test.py]: done

File: rest-api/test/unit_test/api_test/awardee_api_test.py (hunk 6 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/awardee_api_test.py]: done

File: rest-api/tools/import_organizations.py (hunk 7 above)
Comment (Seeker1911) [reply rest-api/tools/import_organizations.py]: think I got it

File: rest-api/tools/import_organizations.py [hunk 9]
```diff
@@ -105,7 +107,7 @@ def _entity_from_row(self, row):
                    row[ORGANIZATION_AWARDEE_ID_COLUMN],
                    row[ORGANIZATION_ORGANIZATION_ID_COLUMN])
       return None
-    return Organization(externalId=row[ORGANIZATION_ORGANIZATION_ID_COLUMN],
+    return Organization(externalId=row[ORGANIZATION_ORGANIZATION_ID_COLUMN].upper(),
```
Comment (danqrodney) [inline rest-api/tools/import_organizations.py]: You're still missing some (e.g. line 104, 129)
